test/apps/iodemo/io_demo.cc

```diff
@@ -453,6 +473,22 @@ class P2pDemoCommon : public UcxContext {
         MemoryPool<SendCompleteCallback>& _pool;
     };
 
+    // Holds details of AM message consisting of several chunks
+    struct UcxAmMsg {
+        UcxAmMsg() : _iov(NULL), _cb(NULL), _count(0)
+        {}
```

Comment:
seems like not needed, then _iov and _cb could be a reference (not a pointer)

Reply:
can you please clarify?

Reply:
as far as I can see you don't use default constructor